src/main_thread/init/multi_thread_content_initializer.ts

```diff
@@ -1094,10 +1094,11 @@ export default class MultiThreadContentInitializer extends ContentInitializer {
           if (this._currentContentInfo?.contentId !== msgData.contentId) {
             return;
           }
-          this._segmentMetrics.resolvers[msgData.value.messageId](
-            msgData.value.segmentSinkMetrics,
-          );
-          delete this._segmentMetrics.resolvers[msgData.value.messageId];
+          const resolveFn = this._segmentMetrics.resolvers[msgData.value.messageId];
+          if (resolveFn !== undefined) {
```

Comment:
I'm OK with the check but in which case could this happen?

If it did happen, this is dangerous as it could mean a leak